app/javascript/components/tooltips/Concept.tsx

```diff
@@ -0,0 +1,423 @@
+  openTooltips.forEach(([openId, openDispatch]) => {
+    if (openId !== id) {
+      openDispatch({
+        id: openId,
+        action: {
+          type: 'hide',
+        },
+      })
+    }
+  })
+}
+
+const shouldShow = (state: TooltipState): boolean => {
+  return state.focus || state.hover || state.requestFocus || state.requestHover
+}
+
+const handleShowRequest = (state: TooltipState): ShowState => {
+  return shouldShow(state) && state.showState === 'hidden'
+    ? 'loading'
+    : state.showState
+}
+
+const handleHideRequest = (state: TooltipState): ShowState => {
+  return !shouldShow(state) ? 'hidden' : state.showState
+}
+
+/**
+ * tooltipReducer
+ * This serves as a state reducer for the Concept (tooltip) Component for use with
+ * the useReducer hook.
+ */
+const tooltipReducer: ShowReducer = function (state, body) {
+  const { id, action } = body
+  const nextState: TooltipState = { ...state }
+
+  // Use action to update the next state
+  switch (action.type) {
+    case 'request-show-from-ref-hover':
+      nextState.requestHover = true && !hasOpenTooltip(id)
+      nextState.showState = handleShowRequest(nextState)
+      break
+    case 'request-hide-from-ref-hover':
+      nextState.requestHover = false
+      nextState.showState = handleHideRequest(nextState)
+      break
+    case 'request-show-from-ref-focus':
+      nextState.requestFocus = true
+      nextState.showState = handleShowRequest(nextState)
+      break
+    case 'request-hide-from-ref-focus':
+      nextState.requestFocus = false
+      nextState.showState = handleHideRequest(nextState)
+      break
+    case 'request-show-hover':
+      nextState.hover = true && !hasOpenTooltip(id)
+      nextState.showState = handleShowRequest(nextState)
+      break
+    case 'request-hide-hover':
+      nextState.hover = false
+      nextState.showState = handleHideRequest(nextState)
+      break
+    case 'request-show-focus':
+      nextState.focus = true
+      nextState.showState = handleShowRequest(nextState)
+      break
+    case 'request-hide-focus':
+      nextState.focus = false
+      nextState.showState = handleHideRequest(nextState)
+      break
+    case 'loaded':
+      nextState.showState = 'invisible'
+      break
+    case 'show':
+      nextState.showState = 'visible'
+      break
+    case 'hide':
+      nextState.showState = 'hidden'
+      break
+    case 'error':
+      nextState.showState = 'error'
+      break
+    default:
+      throw new Error('unhandled tooltip reducer action')
+  }
+
+  return nextState
+}
+
+export const Concept = ({
+  contentEndpoint,
+  hoverRequestToShow,
+  focusRequestToShow,
+  referenceElement,
+  referenceConceptSlug,
+}: ConceptProps): JSX.Element | null => {
+  const tooltipId = `concept-tooltip${
+    referenceConceptSlug ? `-${referenceConceptSlug}` : referenceConceptSlug
+  }`
+
+  // Retrieve the HTML contents from the contentEndpoint
+  const { isLoading, isError, data: html_content } = useQuery<string>(
```

Comment:
Could `html_content` use camelCase?

```suggestion
  const { isLoading, isError, data: htmlContent } = useQuery<string>(
```

Reply:
Yes, I think I just had this separate because of its "unsafe" usage in my head.

Reply:
(it's not unsafe tho :P)

Reply:
fixed